Sources/BrowserServicesKit/SecureVault/SecureVaultManager.swift

```diff
@@ -127,8 +129,16 @@ extension SecureVaultManager: AutofillSecureVaultDelegate {
                                    didRequestCreditCardWithId creditCardId: Int64,
                                    completionHandler: @escaping (SecureVaultModels.CreditCard?) -> Void) {
         do {
-            completionHandler(try SecureVaultFactory.default.makeVault(errorReporter: self.delegate)
-                                .creditCardFor(id: creditCardId))
+            let card = try SecureVaultFactory.default.makeVault(errorReporter: self.delegate).creditCardFor(id: creditCardId)
+
+            delegate?.secureVaultManager(self, didRequestAuthenticationWithCompletionHandler: { authenticated in
```

Comment:
Need to wrap a unit test around this. And I'd also like to quickly look at making the new authentication delegate method optional, and only go through the authentication flow if it's implemented.

Reply:
Took a look at this, an optional protocol is difficult here since it needs to be marked with `@objc` and there are some parameters being used that can't be exposed to the ObjC runtime. Instead, a default implementation could be provided:

```
extension SecureVaultManagerDelegate {
    
    func secureVaultManager(_: SecureVaultManager, didRequestAuthenticationWithCompletionHandler handler: @escaping (Bool) -> Void) {
        handler(true)
    }

}
```

Or, this could just stay a required method - doesn't bother me either way.